src/utils/common/common.c

```diff
@@ -85,8 +85,10 @@ static pthread_mutex_t strerror_r_lock = PTHREAD_MUTEX_INITIALIZER;
 #endif
 
 char *sstrncpy(char *dest, const char *src, size_t n) {
-  strncpy(dest, src, n);
-  dest[n - 1] = '\0';
+  if (n > 0) {
```

Comment:
I went back and forth on this a few time.

* On the one hand, `n == 0` is not really an error and a situation that could naturally occur, e.g. when you're implementing code that appends to the end of a string.
* On the other hand, if we return `NULL` when `n` is zero we can guarantee that we will either return `NULL` or a null terminated string.

Ultimately I decided to go with the stronger guarantee, i.e. `if (n == 0) { return NULL; }`

Reply:
I was undecided on this too.

With modern compiler checks, use of those original functions is much safer, so "safe" wrapper has less value.  What's worse, because compiler does not have any special knowledge about the wrappers, it cannot do similar checks for them.  So for it to actually be safer, it needs to do extra checks.

(One could also add non-null attribute, but as that can check only values that are visible for compiler, it's almost completely useless here.)

> Ultimately I decided to go with the stronger guarantee, i.e. if (n == 0) { return NULL; }

On quick check of  `sstrncpy()` usage (in 1044 places in 167 files), nothing was using its return value, except its test code, so changing return value semantics is fine.